lib/engines/hcktest/hcktest.rb

```diff
@@ -229,20 +230,56 @@ def prepare_tests
         return
       end
 
-      @tests.list_tests(log: true)
+      @test_list = @tests.list_tests(log: true)
     end
 
     def auto_run
       ResourceScope.open do |scope|
-        run_and_configure_setup scope
-        prepare_tests
+        run_studio scope
+        sleep 5 until @studio.up?
+
+        run_tests_without_config
+        run_tests_with_config
 
-        @tests.run
+        @tests.create_project_package
+      end
+    end
+
+    def run_tests_without_config
+      ResourceScope.open do |scope|
+        run_clients_and_configure_setup scope
+        prepare_tests
 
+        @logger.info('Environment ready, running normal tests')
+        @tests.run(@test_list - group_tests_by_config.values.flatten)
         pause_run if @project.options.test.manual
+      end
+    end
 
-        @tests.create_project_package
+    def run_tests_with_config
+      group_tests_by_config.each do |group, tests|
+        next if tests.empty?
+
+        @logger.info("Preparing environment for #{group} tests")
```

Comment:
Log just before running tests instead of logging here because:
- `HCKClient#configure` logs before configuring anyway.
- It is inconsistent with `run_tests_without_config`, which logs just before running tests.